package/kernel/linux/modules/spi.mk

```
@@ -73,3 +73,27 @@ define KernelPackage/spi-dev/description
 endef
 
 $(eval $(call KernelPackage,spi-dev))
+
+
+define KernelPackage/spi-dw-mmio
+  SUBMENU:=$(SPI_MENU)
+  TITLE:=DesignWare SPI controller driver (MMIO)
+  KCONFIG:=\
+    CONFIG_SPI=y \
+    CONFIG_SPI_DESIGNWARE \
+    CONFIG_SPI_DW_DMA=n \
+    CONFIG_SPI_DW_MMIO=y \
+    CONFIG_SPI_DW_PCI=n \
+    CONFIG_SPI_DYNAMIC=y \
+    CONFIG_SPI_MASTER=y
+  FILES:=\
+    $(LINUX_DIR)/drivers/spi/spi-dw.ko \
+    $(LINUX_DIR)/drivers/spi/spi-dw-mmio.ko
+  AUTOLOAD:=$(call AutoLoad,spi-dw-mmio)
```

```suggestion
  AUTOLOAD:=$(call AutoProbe,spi-dw-mmio)
```
This should fix loading the module.

`AutoLoad` needs a order as second parameter, `AutoProbe` will automatically load all dependencies and is preferred.

---

Thanks for the explanation! I just changed it.